crates/pixi_build_frontend/src/into_build_types.rs

```diff
@@ -0,0 +1,180 @@
+//! these are used to convert the `pixi_spec` types to the `pixi_build_types` types
+//! we want to keep the conversion here, as we do not want `pixi_build_types` to depend on `pixi_spec`
+//!
+//! This will mostly be boilerplate conversions but some of these are a bit more interesting
+
+// Namespace to pbt, *please use* exclusively so we do not get confused between the two different types
+use indexmap::IndexMap;
+use pixi_build_types as pbt;
+use pixi_manifest::{PackageManifest, PackageTarget, TargetSelector, Targets};
+use pixi_spec::{PixiSpec, Reference};
+use rattler_conda_types::PackageName;
+
+/// Conversion from a `PixiSpec` to a `pbt::PixiSpecV1`.
+fn to_pixi_spec_v1(spec: &PixiSpec) -> pbt::PixiSpecV1 {
+    match spec {
+        // We consider both `Version` and `DetailedVersion` to use
+        // the NamelessMatchSpecV1 variant
+        PixiSpec::Version(version_spec) => pbt::PixiSpecV1::DetailedVersion(version_spec.into()),
+        PixiSpec::DetailedVersion(detailed_spec) => {
+            pbt::PixiSpecV1::DetailedVersion(pbt::NamelessMatchSpecV1 {
+                version: detailed_spec.version.clone(),
+                build: detailed_spec.build.clone(),
+                build_number: detailed_spec.build_number.clone(),
+                file_name: detailed_spec.file_name.clone(),
+                channel: detailed_spec
+                    .channel
+                    .as_ref()
+                    .map(|c| c.to_string())
+                    .clone(),
+                subdir: detailed_spec.subdir.clone(),
+                namespace: None,
+                md5: detailed_spec.md5.map(Into::into),
+                sha256: detailed_spec.sha256.map(Into::into),
+                url: None,
+            })
+        }
+        PixiSpec::Url(url_spec) => pbt::PixiSpecV1::Url(pbt::UrlSpecV1 {
+            url: url_spec.url.clone(),
+            md5: url_spec.md5.map(Into::into),
+            sha256: url_spec.sha256.map(Into::into),
+        }),
+        PixiSpec::Git(git_spec) => pbt::PixiSpecV1::Git(pbt::GitSpecV1 {
+            git: git_spec.git.clone(),
+            rev: git_spec.rev.clone().map(|r| match r {
+                Reference::Branch(b) => pbt::GitReferenceV1::Branch(b.clone()),
+                Reference::Tag(t) => pbt::GitReferenceV1::Tag(t.clone()),
+                Reference::Rev(rev) => pbt::GitReferenceV1::Rev(rev.clone()),
+                Reference::DefaultBranch => pbt::GitReferenceV1::DefaultBranch,
+            }),
+            subdirectory: git_spec.subdirectory.clone(),
+        }),
+        PixiSpec::Path(path_spec) => pbt::PixiSpecV1::Path(pbt::PathSpecV1 {
+            path: path_spec.path.to_string(),
+        }),
+    }
+}
+
+/// Converts an iterator of `PackageName` and `PixiSpec` to a `IndexMap<String, pbt::PixiSpecV1>`.
+fn to_pbt_dependencies<'a>(
+    iter: impl Iterator<Item = (&'a PackageName, &'a PixiSpec)>,
+) -> IndexMap<String, pbt::PixiSpecV1> {
```

Comment:
Should we have a type/alias for a SourcePackageName, to make sure we dont mix this up later?